src/observers/WarningMessageObserver.ts

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { MessageItem, vscode } from '../vscodeAdapter';
+import { Scheduler, Subject } from 'rx';
+import { BaseEvent, OmnisharpServerOnError, OmnisharpServerMsBuildProjectDiagnostics } from "../omnisharp/loggingEvents";
+
+export interface MessageItemWithCommand extends MessageItem {
+    command: string;
+}
+
+export class WarningMessageObserver {
+    private warningMessageDebouncer: Subject<BaseEvent>;
+
+    constructor(private vscode: vscode, scheduler?: Scheduler) {
+        this.warningMessageDebouncer = new Subject<BaseEvent>();
+        this.warningMessageDebouncer.debounce(1500, scheduler).subscribe(async event => {
```

Comment:
We've gotten feedback that having a ton of these messages appear is annoying. The command we have here doesn't appear to scroll the output to a specific location. Instead of "Debounce", should we change this observer to just raise one notification if it gets that event?

Reply:
But if the server restarts then ?

Reply:
Ah, this was me misunderstanding debouce. Looks good to me.